### PR TITLE
Support segmentation masks in CutMix layer 

### PR DESCRIPTION
**Pull request by soma2000-lang**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks for the PR!

As-is I think this may not work the way we expect it to (see my comment about randomization and calling `self._cutmix` twice)

Could you please create a Colab or example script like the one in #1584 which demonstrates this layer in action on segmentation masks? Then we can visually and numerically verify that it's doing what we'd expect.

**Comment by soma2000-lang** [comment]
Sure @ianstenbit  I have been already working on a colab

File: keras_cv/layers/preprocessing/cut_mix.py [hunk 1]
```diff
@@ -11,32 +11,37 @@
...
 
 
-@keras.utils.register_keras_serializable(package="keras_cv")
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
 class CutMix(BaseImageAugmentationLayer):
     """CutMix implements the CutMix data augmentation technique.
 
     Args:
+
+        alpha: Float between 0 and 1.  Inverse scale parameter for the gamma
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/cut_mix.py:28]: It looks like this is now duplicated in the docstring

File: keras_cv/layers/preprocessing/cut_mix.py [hunk 2]
```diff
@@ -11,32 +11,37 @@
...
-from tensorflow import keras
 
 from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
     BaseImageAugmentationLayer,
 )
 from keras_cv.utils import fill_utils
 
 
-@keras.utils.register_keras_serializable(package="keras_cv")
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/cut_mix.py:22]: Let's import `keras` and use it here

Comment (soma2000-lang) [reply keras_cv/layers/preprocessing/cut_mix.py:22]: @ianstenbit Sure will make those changes!

File: keras_cv/layers/preprocessing/cut_mix.py [hunk 3]
```diff
@@ -63,7 +68,7 @@ def _batch_augment(self, inputs):
         self._validate_inputs(inputs)
         images = inputs.get("images", None)
         labels = inputs.get("labels", None)
-
+        segmentation_masks = inputs.get("segmentation_masks", None)
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/cut_mix.py:71]: lets add the newline back one before images

File: keras_cv/layers/preprocessing/cut_mix.py [hunk 4]
```diff
@@ -73,13 +71,18 @@ def _batch_augment(self, inputs):
         images, labels = self._update_labels(*self._cutmix(images, labels))
         inputs["images"] = images
         inputs["labels"] = labels
+        if segmentation_masks is not None:
+            segmentation_masks = self._update_segmentation_masks(
+                *self._cutmix(images, labels, segmentation_masks)
```
[the remark this answers is not in the document]
Comment (LukeWood) [reply keras_cv/layers/preprocessing/cut_mix.py:83]: +1, you'll need to call it once, then call the update methods accordingly.

**Comment by cosmo3769** [comment]
Hey @soma2000-lang, are you working on this?
